onnxruntime/core/profile/context.h

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <thread>
+#include <mutex>
+#include <string>
+#include <unordered_map>
+
+namespace onnxruntime {
+namespace profile {
+
+class Context {
```

Comment:
>Context [](start = 6, length = 7)

would this class be equivalent to a thread_local string?
on the other hand, it seems that thread_local variable usage can be complicated...

Reply:
I feel this design is more extensible. It's like a centralized information manager. If we choose thread local variable, how do we pass it from `training_runner.cc` into `sequential_executor.cc`? As you expected, it'd be more complicated. 

---
In reply to: [428994850](https://github.com/microsoft/onnxruntime/pull/4013#discussion_r428994850) [](ancestors = 428994850)

Reply:
I think something like this could work:
```
struct ContextState {
  string tag;
  // can add more stuff here
};

ContextState& GetThreadLocalContextState() {
  thread_local ContextState c{};
  return c;
}
```
This is aligned with the thread_local guidelines here: https://google.github.io/styleguide/cppguide.html#thread_local

Possible consideration for a future change.